app/helpers/reactjs_helper.rb

```diff
@@ -24,7 +21,25 @@ def webpacked_plugins_with_global_js
   end
 
   def webpacked_plugins_css_for(*plugin_names)
-    css_tags_for(select_requested_plugins(plugin_names)).join.html_safe
+    Foreman::Deprecation.deprecation_warning('3.9', '`webpacked_plugins_css_for` is deprecated, plugin css is already loaded.')
+  end
+
+  def read_webpack_manifest
+    root = File.expand_path(File.dirname(__FILE__) + "/../..")
+    file = File.read(root + '/public/webpack/manifest.json')
+    JSON.parse(file)
+  end
+
+  def get_webpack_foreman_vendor_js
+    data = read_webpack_manifest
+    foreman_vendor_js = data['assetsByChunkName']['foreman-vendor'].find { |value| value.end_with?('.js') }
```

Comment:
This still reads the manifest on every request, possibly multiple times. Would it make sense to cache this?

Reply:
@ekohl Yeah, for how long?